### PR TITLE
ansi-c: use ranged for

### PR DESCRIPTION
**Pull request by kroening**
For better readability!

- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/ansi-c/c_typecheck_base.cpp [hunk 1]
```diff
@@ -148,11 +148,8 @@ void c_typecheck_baset::typecheck_new_symbol(symbolt &symbol)
     {
       // we don't need the identifiers
       code_typet &code_type=to_code_type(symbol.type);
-      for(code_typet::parameterst::iterator
-          it=code_type.parameters().begin();
-          it!=code_type.parameters().end();
-          it++)
-        it->set_identifier(irep_idt());
+      for(auto &parameter : code_type.parameters())
+        parameter.set_identifier(irep_idt());
```
Comment (tautschnig) [inline src/ansi-c/c_typecheck_base.cpp:153]: In a second commit in this PR maybe get rid of the `code_type` named temporary? And the block around these lines?

**Review by tautschnig — APPROVED** [review]

File: src/ansi-c/c_typecheck_base.cpp (hunk 1 above)
Comment (kroening) [reply src/ansi-c/c_typecheck_base.cpp:153]: done

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 149dc1f9bf0bba39177c9fbb29acb194c2a4417f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96631034
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 149dc1f9bf0bba39177c9fbb29acb194c2a4417f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96631034

**Pull request merged**